src/core/environmentAdapter.ts

```diff
@@ -6,11 +6,12 @@ export function init(implementation: EnvironmentAdapter): void {
 }
 
 export interface ChildProcess { // subset of child_process.ChildProcess
-    // TODO: add some stuff like in child_process.ChildProcess
+    on(event: 'exit', listener: (code: number) => void): this;
+    // TODO: add more stuff like in child_process.ChildProcess
 }
 
 export interface EnvironmentAdapter {
     getEnvironmentName(): 'electron'|'browser'|'vscode'
-    runShellCommand(command: string): ChildProcess
+    runShellCommand(command: string, options: unknown): ChildProcess
```

Comment:
I find it important to specify what options can be used and should be implemented:
```suggestion
    runShellCommand(command: string, options?: { cwd?: string }): ChildProcess
```
or more speaking, but not sure if too long:
```suggestion
    runShellCommand(command: string, options?: { currentWorkingDirectory?: string }): ChildProcess
```